monai/transforms/smooth_field/array.py

```diff
@@ -14,90 +14,163 @@
+        self.pad = pad
+        self.low = low
+        self.high = high
+        self.channels = channels
+        self.mode = mode
+        self.align_corners = align_corners
+        self.device = device
+
+        self.spatial_size: Optional[Sequence[int]] = None
+        self.spatial_zoom: Optional[Sequence[float]] = None
+
+        if low >= high:
+            raise ValueError("Value for `low` must be less than `high` otherwise field will be zeros")
+
+        self.total_rand_size = tuple(rs + self.pad * 2 for rs in self.rand_size)
+
+        self.field = torch.ones((1, self.channels) + self.total_rand_size, device=self.device) * pad_val
+
+        self.crand_size = (self.channels,) + self.rand_size
+
+        pad_slice = slice(None) if self.pad == 0 else slice(self.pad, -self.pad)
+        self.rand_slices = (0, slice(None)) + (pad_slice,) * len(self.rand_size)
+
+        self.set_spatial_size(spatial_size)
 
     def randomize(self, data: Optional[Any] = None) -> None:
-        self.field = self.R.uniform(self.low, self.high, (self.channels,) + self.rand_size)  # type: ignore
-        if self.padder is not None:
-            self.field = self.padder(self.field)
+        self.field[self.rand_slices] = torch.from_numpy(self.R.uniform(self.low, self.high, self.crand_size))
+
+    def set_spatial_size(self, spatial_size: Optional[Sequence[int]]) -> None:
+        """
+        Set the `spatial_size` and `spatial_zoom` attributes used for interpolating the field to the given
+        dimension, or not interpolate at all if None.
+
+        Args:
+            spatial_size: new size to interpolate to, or None to not interpolate
+        """
+        if spatial_size is None:
+            self.spatial_size = None
+            self.spatial_zoom = None
+        else:
+            self.spatial_size = tuple(spatial_size)
+            self.spatial_zoom = tuple(s / f for s, f in zip(self.spatial_size, self.total_rand_size))
+
+    def set_mode(self, mode: Union[monai.utils.InterpolateMode, str]) -> None:
+        self.mode = mode
+
+    def __call__(self, randomize=False) -> torch.Tensor:
+        if randomize:
+            self.randomize()
+
+        field = self.field.to(self.device).clone()
 
-    def __call__(self):
-        resized_field = self.resizer(self.field)
+        if self.spatial_zoom is not None:
+            resized_field = interpolate(  # type: ignore
+                input=field,  # type: ignore
+                scale_factor=self.spatial_zoom,
+                mode=look_up_option(self.mode, InterpolateMode).value,
+                align_corners=self.align_corners,
+                recompute_scale_factor=False,
+            )
 
-        return rescale_array(resized_field, self.field.min(), self.field.max())
+            mina = resized_field.min()
+            maxa = resized_field.max()
+            minv = self.field.min()
+            maxv = self.field.max()
+
+            # faster than rescale_array (?)
```

Comment:
May I know why it's faster than `rescale_array`?
https://github.com/Project-MONAI/MONAI/blob/dev/monai/transforms/utils.py#L151

Thanks.

Reply:
`rescale_array` rescales the input to the unit range then again to the requested range and has various checks, this avoids a lot of unneeded computation, and uses the inline Pytorch operators to avoid needless copying. Like the previous mentioned issue we're deviating from standard here for performance reasons so the question is if it's worth it.

Reply:
OK, BTW, do you think it's possible to enhance the `rescale_array` for better performance as you mentioned?
I think that would be also useful for other components.

Thanks.

Reply:
`rescale_array` is first rescaling to unit range then to the desired range so it's doing something slightly different. It also has to check that the input's range is not zero but that's checked here in the constructor, so the time savings is somewhat particular to this application.